internal/reconstitution/queueprocessor.go

```diff
@@ -30,7 +30,7 @@ func (q *queueProcessor) processQueueItem(ctx context.Context) bool {
 	}
 	defer q.Queue.Done(item)
 
-	req, ok := item.(*Request)
+	req, ok := item.(Request)
```

Comment:
Kind of unrelated change: using pointers breaks the workqueues de-duplication since it relies on equality checks.